requests_toolbelt/downloadutils/stream.py

```diff
@@ -122,8 +138,19 @@ def stream_response_to_file(response, path=None):
             )
         fd = open(filename, 'wb')
 
-    for chunk in response.iter_content(chunk_size=512):
+    try:
+        bytes_total = int(response.headers.get('content-length', None))
+    except TypeError:
+        bytes_total = None  # not specified or invalid
```

Comment:
If it's not specified than it was probably sent using `Transfer-Encoding: chunked` which is not invalid.

This should be done differently, *if* at all.